src/main/java/edu/cornell/kfs/sys/service/impl/BatchFileDirectoryServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package edu.cornell.kfs.sys.service.impl;
+
+import edu.cornell.kfs.sys.service.BatchFileDirectoryService;
+import edu.cornell.kfs.sys.util.SubDirectoryWalker;
+import org.kuali.rice.core.api.util.KeyValue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchFileDirectoryServiceImpl implements BatchFileDirectoryService {
+
+    @Override
+    public List<KeyValue> buildDirectoryKeyValuesList(List<File> rootDirectories) {
```

Comment:
Why? Are you concerned that some implementations may be cacheable and some may not be? I saw other places (mostly Rice I think), where the annotation was on the interface, so I followed that pattern here.